packages/core/src/createActor.ts

```diff
@@ -207,7 +211,23 @@ export class Actor<TLogic extends AnyActorLogic>
       this.system._set(systemId, this);
     }
 
-    this._initState(options?.snapshot ?? options?.state);
+    this._synchronizer = options?.sync;
+
+    const initialSnapshot =
+      this._synchronizer?.getSnapshot() ?? options?.snapshot ?? options?.state;
+
+    this._initState(initialSnapshot);
+
+    if (this._synchronizer) {
+      this._synchronizerSubscription = this._synchronizer.subscribe(
+        (rawSnapshot) => {
+          const restoredSnapshot =
+            this.logic.restoreSnapshot?.(rawSnapshot, this._actorScope) ??
+            rawSnapshot;
+          this.update(restoredSnapshot, { type: '@xstate.sync' });
```

Comment:
We don't use `@` for other builtin events like init, done.invoke and more
```suggestion
          this.update(restoredSnapshot, { type: 'xstate.sync' });
```